mersenne-31/src/poseidon2.rs

```diff
@@ -35,18 +35,21 @@ pub const MATRIX_DIAG_16_MERSENNE31: [Mersenne31; 16] =
 
 // We make use of the fact that most entries are a power of 2.
 // Note that this array is 1 element shorter than MATRIX_DIAG_16_MERSENNE31 as we do not include the first element.
-const MATRIX_DIAG_16_MONTY_SHIFTS: [i32; 15] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 10, 12, 13, 14, 15, 16];
+const MATRIX_DIAG_16_MONTY_SHIFTS: [u64; 15] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 10, 12, 13, 14, 15, 16];
```

Comment:
Could use a smaller type like `u8`? It probably shouldn't matter since the compiler should do range analysis anyway, but I don't see a downside (`u64` implements `Shl<u8>` etc. so no conversions)

Reply:
Should this also be renamed, since it's M31 there's no Monty logic right?

Reply:
Whoops, yeah good catch should definitely not have MONTY in the name. Will change it to: POSEIDON2_INTERNAL_MATRIX_DIAG_16_SHIFTS.

Happy to change the type as well. I put it at u64 basically just to match with the mul_2exp_u64 function but there is no real need for that.